File: c/func.c
int foo() {
  return 42;
}

void bar(int *x) {
  *x = 21;
}

void baz(int *x, long n, int *y) {
    int sum = 0;
    for (int i = 0; i < n; i++)
        sum += y[i];
    *x = sum;
}

void rust_fadd(const float *a, const float *b, float *out) {
    *out = *a + *b;
}

File: func.rs
#[no_mangle]
pub fn foo() -> i32 {
  42
}

#[no_mangle]
pub fn bar(x: &mut i32) {
	*x = 21;
}

#[no_mangle]
pub fn baz(x: &mut i32, y: &[i32]) {
    *x = y.iter().sum();
}

#[no_mangle]
pub fn rust_fadd(a: &f32, b: &f32, out: &mut f32) {
    *out = a + b;
}
